wptherml/em.py

```diff
@@ -955,65 +966,208 @@ def compute_stpv_gradient(self):
-            _T += 1
-            if P_emit > P_abs:
-                _kill = 0
-        
-        self._compute_pv_stpv_power_density(self.wavelength_array)
-        # reverse stack again and add active layer and get absorbed power into the structure
-        self.reverse_stack()
-
-        # approximate ideal spectral response assuming \lambda_bg = 700 nm
-        self.lambda_bandgap = 700e-9
-        self.spectral_response = self.wavelength_array / self.lambda_bandgap
-        # make sure we have the solar spectrum
+    
+        #while(_kill):
+        #    _bbs = self._compute_blackbody_spectrum(self.wavelength_array, _T)
+        #    P_emit = np.trapz( np.pi/2 * _bbs * (emissivity_A_B + absorptivity_AB_T), self.wavelength_array)
+        #    _T += 1
+        #    if P_emit > absorptivity_B_T :
+        #        _kill = 0
+
+        # assign _T to self.temperature
+        self.temperature = _T 
+
+
+
+    def compute_pv_stpv_jsc_gradient(self):
+        """
+        Computes the gradient of the pv-stpv short circuit current as defined 
+
+        Attributes
+        ----------
+
+        e_gradient_index : Integer
+             Length of the emissivity gradient array.
+
+        emissivity_gradient_array_prime : Array
+            (Emissivity gradient array x Wavelength array) / Lambda bandgap.
+
+        pv_stpv_short_circuit_current_gradient : 1 x e_gradient_index numpy array of floats
+            Derivatives of the hort circuit current as defined in Eq. (45) here: https://www.overleaf.com/project/648a0cfeae29e31e10afc075
+        None
+        """
+
+        # Acquire necessary variables
         self._solar_spectrum = self._read_AM()
-        # now compute pv_stpv short circuit current
+        self.compute_spectrum_gradient()
 
-        self._compute_pv_short_circuit_current(
-            self.wavelength_array,
-            self.emissivity_array,
-            self.spectral_response,
-            self._solar_spectrum,
-        )
+        # Initialize short circuit current array
+        e_gradient_index = len(self.emissivity_gradient_array[0, :])
+        self.pv_stpv_short_circuit_current_gradient = np.zeros(e_gradient_index)
+
+        bg_idx = np.abs(self.wavelength_array - self.pv_lambda_bandgap).argmin()
+        _spectral_response = np.zeros_like(self.wavelength_array)
+
+        _spectral_response[:bg_idx] = self.wavelength_array[:bg_idx] / self.pv_lambda_bandgap
+
+        # Iterate over material thicknesses
+        for i in range(0, e_gradient_index):
+            self.pv_stpv_short_circuit_current_gradient[i] = np.trapz(
+                self.emissivity_gradient_array[:,i]
+                * _spectral_response
+                * self._solar_spectrum,
+                self.wavelength_array,
+            )  # Integrate for short circuit current
+
+    
+    def compute_pv_stpv_splitting_power(self):
 
 
+    def compute_pv_stpv_total_incident_power(self):
```

Comment:
This will be proportional to the short circuit current, and will also require information about the Fill Factor and Open-Circuit Voltage.  We do have that data, but let's leave dealing with this for later.